### PR TITLE
Add folder commands

### PR DESCRIPTION
**Pull request by openorclose**

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -86,6 +86,7 @@ Format: `*add* u/URL n/NAME [f/FOLDER] [t/TAG]…​`
 
 ****
 * A bookmark can have any number of tags (including 0).
+* `FOLDER` must be an existing folder (see <<folder>>.
```
Comment (dorcastan) [inline docs/UserGuide.adoc]:
```suggestion
* `FOLDER` must be an existing folder (see <<folder>>).
```
Where does the link `folder` go to?

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
 * `*add* u/nus-cs2103-ay1920s1.github.io n/Module Website f/CS2103T`
 * `*add* u/www.youtube.com/watch?v=9AMcN-wkspU n/IntelliJ Tips and Tricks t/video t/watch-later`
 
+=== Adding a new folder *`newfolder`*
```
Comment (dorcastan) [inline docs/UserGuide.adoc]:
```suggestion
=== Adding a new folder *`folder`*
```
To match similar commands like `annotate` and `reminder`

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
...
+
+Creates a new folder for organizing bookmarks into.
+
+Format: `*newfolder* FOLDER_NAME [p/PARENT_FOLDER]`
+
+****
+* `FOLDER_NAME` must not have already existed.
```
Comment (dorcastan) [inline docs/UserGuide.adoc]:
```suggestion
* `FOLDER_NAME` must not already exist.
```

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
...
+* `PARENT_FOLDER` must be an existing folder.
+* If `PARENT_FOLDER` is not specified, `FOLDER_NAME` will have no parent, and be at the root.
+****
+
+Examples:
+
+* `*newfolder* CS2103 p/NUS +
+Creates a new folder named `CS2103` under the parent folder `NUS`.
+
+=== Deleting a folder *`deletefolder`*
```
Comment (dorcastan) [inline docs/UserGuide.adoc]:
```suggestion
=== Deleting a folder *`folder delete`*
```
Again, to match similar commands. Should we also add a `folder edit`?

File: docs/UserGuide.adoc [hunk 5]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
...
+
+****
+* `FOLDER_NAME` must be an existing folder.
+* If `FOLDER_NAME` contains other folders, those folders will be deleted as well.
+****
+
+Examples:
+
+* `*deletefolder* CS2103 +
+Deletes the folder named `CS2103`.
```
Comment (dorcastan) [inline docs/UserGuide.adoc]:
```suggestion
Deletes the folder named `CS2103` and all of its subfolders.
```

File: docs/UserGuide.adoc [hunk 6]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
 * `*add* u/nus-cs2103-ay1920s1.github.io n/Module Website f/CS2103T`
 * `*add* u/www.youtube.com/watch?v=9AMcN-wkspU n/IntelliJ Tips and Tricks t/video t/watch-later`
 
+=== Adding a new folder *`newfolder`*
+
+Creates a new folder for organizing bookmarks into.
+
+Format: `*newfolder* FOLDER_NAME [p/PARENT_FOLDER]`
```
Comment (dorcastan) [inline docs/UserGuide.adoc]:
```suggestion
Format: `*folder* FOLDER_NAME [p/PARENT_FOLDER]`
```

File: docs/UserGuide.adoc [hunk 7]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
 * `*add* u/nus-cs2103-ay1920s1.github.io n/Module Website f/CS2103T`
 * `*add* u/www.youtube.com/watch?v=9AMcN-wkspU n/IntelliJ Tips and Tricks t/video t/watch-later`
 
+=== Adding a new folder *`folder`*
+
+Creates a new folder for organizing bookmarks into.
+
+Format: `*newfolder* FOLDER_NAME [p/PARENT_FOLDER]`
```
Comment (kxyoke) [inline docs/UserGuide.adoc]:
```suggestion
Format: `*folder* FOLDER_NAME [p/PARENT_FOLDER]`
```

File: docs/UserGuide.adoc [hunk 8]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
...
+
+****
+* `FOLDER_NAME` must not already exist.
+* `PARENT_FOLDER` must be an existing folder.
+* If `PARENT_FOLDER` is not specified, `FOLDER_NAME` will have no parent, and be at the root.
```
Comment (kxyoke) [inline docs/UserGuide.adoc]:
```suggestion
* If `PARENT_FOLDER` is not specified, `FOLDER_NAME` will have no parent and will be placed at the root.
```

File: docs/UserGuide.adoc [hunk 9]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
...
+****
+
+Examples:
+
+* `*newfolder* CS2103 p/NUS +
```
Comment (kxyoke) [inline docs/UserGuide.adoc]:
```suggestion
* `*folder* CS2103 p/NUS +
```

File: docs/UserGuide.adoc [hunk 10]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
...
+Creates a new folder named `CS2103` under the parent folder `NUS`.
+
+=== Deleting a folder *`folder delete`*
+
+Creates a new folder for organizing bookmarks into.
+
+Format: `*deletefolder* FOLDER_NAME`
```
Comment (kxyoke) [inline docs/UserGuide.adoc]:
```suggestion
Format: `*folder delete* FOLDER_NAME`
```

File: docs/UserGuide.adoc [hunk 11]
```diff
@@ -98,6 +99,39 @@ Only valid if a web-page is currently being viewed.
...
+
+****
+* `FOLDER_NAME` must be an existing folder.
+* If `FOLDER_NAME` contains other folders, those folders will be deleted as well.
+****
+
+Examples:
+
+* `*deletefolder* CS2103 +
```
Comment (kxyoke) [inline docs/UserGuide.adoc]:
```suggestion
* `*folder delete* CS2103 +
```

File: docs/UserGuide.adoc (hunk 8 above)
Comment (kxyoke) [reply docs/UserGuide.adoc]: This I'm not sure which would be better, but personally I feel it's easier to read with less comma splicing

Comment (openorclose) [reply docs/UserGuide.adoc]: ok!

**Pull request merged**